x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/get_threat_list.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { getQueryFilter } from '../../../../../common/detection_engine/get_query_filter';
+import {
+  GetSortWithTieBreakerOptions,
+  GetThreatListOptions,
+  SortWithTieBreaker,
+  ThreatListItem,
+} from './types';
+
+/**
+ * This should not exceed 10000 (10k)
+ */
+export const MAX_PER_PAGE = 9000;
+
+export const getThreatList = async ({
+  callCluster,
+  query,
+  index,
+  perPage,
+  searchAfter,
+  sortField,
+  sortOrder,
+  exceptionItems,
+  threatFilters,
+}: GetThreatListOptions): Promise<SearchResponse<ThreatListItem>> => {
+  const calculatedPerPage = perPage ?? MAX_PER_PAGE;
+  if (calculatedPerPage > 10000) {
+    throw new TypeError('perPage cannot exceed the size of 10000');
+  }
+  const queryFilter = getQueryFilter(query, 'kuery', threatFilters, index, exceptionItems);
+  const response: SearchResponse<ThreatListItem> = await callCluster('search', {
+    body: {
+      query: queryFilter,
+      search_after: searchAfter,
+      sort: getSortWithTieBreaker({ sortField, sortOrder }),
+    },
+    ignoreUnavailable: true,
+    index,
+    size: calculatedPerPage,
+  });
+  return response;
+};
+
+export const getSortWithTieBreaker = ({
+  sortField,
+  sortOrder,
+}: GetSortWithTieBreakerOptions): SortWithTieBreaker[] => {
+  const ascOrDesc = sortOrder ?? 'asc';
+  if (sortField != null) {
+    return [{ [sortField]: ascOrDesc, '@timestamp': 'asc' }];
```

Comment:
Not sure if this comment is at all relevant - will threat list rules have timestamp overide? If so, would that matter at all here? Like wanting to use that field over `@timestamp`?

Reply:
It can have an override of a timestamp. That is a good point. I will add this to a small TODO list I am tracking. I will more than likely have to revisit this item in a follow up PR. Appreciate you finding this.